staging/src/k8s.io/component-base/config/testing/roundtrip.go

```diff
@@ -90,3 +92,89 @@ func GetRoundtripTestCases(scheme *runtime.Scheme, disallowMarshalGroupVersions
 	}
 	return cases
 }
+
+func runRoundTripTests(t *testing.T, tests []TestCase, codecs serializer.CodecFactory) {
```

Comment:
@liggitt do we already have tests for some of the more general things checked here, like "deep copies and encodes don't mutate the original object" and "encoding is stable"? I'd expect some coverage in the basic apimachinery, or something. They sound like good things to test, but if we already have coverage (ideally fuzzed) then maybe we only need to check the round-trip result here, rather than the general things too.